src/os/FileJournal.cc

```diff
@@ -866,6 +871,45 @@ int FileJournal::check_for_full(uint64_t seq, off64_t pos, off64_t size)
   return -ENOSPC;
 }
 
+int FileJournal::prepare_padding_entry(bufferlist& bl, off64_t& queue_pos)
```

Comment:
It seems like a much simpler approach would be:

change prepare_multi_write to pass the start alignment and bool is_last to prepare_single_write.  Then make prepare_single_write only pad if (1) the write is big and we care about alignment of the data payload (say, > 16KB), or (2) we are the last entry in the batch, in which case we need to add tail padding to align ourselves.

What do you think?